### DISCUSSION TITLE
Unable to load legacy in openssl 3.1.1  for iOS project

### DISCUSSION THREAD

**Discussion by khusboo7** [openssl/openssl#21648] [Q&A] [answered]
I have updated the openssl version in my iOS project from 1.1.1t to 3.1.1. Since I want to use the older algorithm for encryption of pkcs12, so I tried the loading the legacy but its not loading. It is returning nil. Below is the code snippet which I tried:
```
OSSL_PROVIDER *legacy = OSSL_PROVIDER_try_load(NULL, "legacy", 1);
OSSL_PROVIDER *defaultProvider = OSSL_PROVIDER_try_load(NULL, "default", 1);
int nid_key = NID_pbe_WithSHA1And3_Key_TripleDES_CBC;
int nid_cert = NID_pbe_WithSHA1And40BitRC2_CBC;
```
Also, I tried following code and its is returning `nil`
`OSSL_PROVIDER_available(NULL, "legacy")`

**Comment by slontis** [comment, marked as answer]
Refer to the documentation for  OSSL_PROVIDER_load() in doc/man3/OSSL_PROVIDER.pod.
which discusses the "MODULESDIR" directory and the environment variable OPENSSL_MODULES.

**Comment by khusboo7** [comment]
Closing this,as the issue is resolve.
